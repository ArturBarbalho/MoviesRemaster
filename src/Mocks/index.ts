import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }