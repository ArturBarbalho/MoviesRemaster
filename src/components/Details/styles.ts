import styled from 'styled-components'
import media from 'styled-media-query'
export const Wrapper = styled.main`
background-color:black;
@media(max-width:400px){
    height: 140vh;
}
`
export const Info = styled.div`
max-width:1000px;
${media.greaterThan('large')`
margin-left:50px;
`}
${media.lessThan('small')`
text-align:center;
`}
p{
    display:flex;
    font-size:22px;
    color:white;
    margin-bottom:30px;
}
${media.lessThan('small')`
p{
    justify-content:center;
}
`}
span{
    margin-right:10px;
}
`

export const Content = styled.div`
position:absolute;
display:flex;
justify-content:center;
align-items:center;
width:100%;
height:100vh;
text-shadow:2px 5px 5px black;
${media.lessThan('medium')`
    flex-direction:column;
    justify-content:space-around;
`}
@media(max-width:400px){
    margin-top:100px;
}
`

export const Gen = styled.div`
font-size:22px;
display: flex;
align-items:center;
${media.lessThan('small')`
justify-content:center;
`}
`