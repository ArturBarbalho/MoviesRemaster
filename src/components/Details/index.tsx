import { DetailsProps } from '../../typesProps'
import Genres from '../Genres'
import Image from '../Image'
import Overview from '../Overview'
import * as S from './styles'
type props = {
  props:DetailsProps
}
const Details = ({props}:props) => (
  <S.Wrapper>
     <S.Content>
      <Image size='medium' src={props.smallsrc} />
      <S.Info>
        <p> <span>Relase date:</span> {props.relase}</p>
        <p><span>Vote avarage:</span> {props.vote_avarage}</p>
        <S.Gen>
        <p><span style={{marginRight:'0'}} >Genres: </span></p>
        <Genres genres={props.genres} size='details'/> 
        </S.Gen> 
        <Overview overview={props.overview} size='small' />
      </S.Info>
    </S.Content>
    <Image src={props.bigsrc} size='large' />
  </S.Wrapper>
)

export default Details