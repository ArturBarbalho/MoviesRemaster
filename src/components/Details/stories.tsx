import { Story, Meta } from '@storybook/react/types-6-0'
import Details from '.'
import { DetailsMock } from '../../Mocks'

export default {
  title: 'Details',
  component: Details
} as Meta

export const Default: Story = () => <Details props={DetailsMock} />