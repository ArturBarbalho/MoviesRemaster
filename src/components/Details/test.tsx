import { render, screen } from '@testing-library/react'

import Details from '.'
import { DetailsMock } from '../../Mocks'

describe('<Details />', () => {
  it('should render the correct text', () => {
    const { container } = render(<Details props={DetailsMock} />)

    expect(screen.getByText('2008-04-14')).toBeInTheDocument()
    expect(screen.getByText('5')).toBeInTheDocument()
    

    expect(container.firstChild).toMatchSnapshot()
  })
})