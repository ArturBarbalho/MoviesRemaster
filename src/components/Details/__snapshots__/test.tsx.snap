// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render the correct text 1`] = `
<main
  class="styles__Wrapper-sc-yoymcz-0 iySqxh"
>
  <div
    class="styles__Content-sc-yoymcz-2 jioCrE"
  >
    <img
      class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
      src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
    />
    <div
      class="styles__Info-sc-yoymcz-1 Suydv"
    >
      <p>
         
        <span>
          Relase date:
        </span>
         
        2008-04-14
      </p>
      <p>
        <span>
          Vote avarage:
        </span>
         
        5
      </p>
      <div
        class="styles__Gen-sc-yoymcz-3 fZJhvW"
      >
        <p>
          <span
            style="margin-right: 0px;"
          >
            Genres: 
          </span>
        </p>
        <div
          class="styles__Wrapper-sc-6y5ihb-1 hOLMLT"
        >
          <div
            class="styles__Genre-sc-6y5ihb-0 hhqlsD"
          >
            Action
          </div>
          <div
            class="styles__Genre-sc-6y5ihb-0 hhqlsD"
          >
            Comedy
          </div>
        </div>
      </div>
      <p
        class="styles__Wrapper-sc-mhbed0-0 ldjcec"
      >
        Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
      </p>
    </div>
  </div>
  <img
    class="styles__Wrapper-sc-y6e3xk-0 kFeMay"
    src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
  />
</main>
`;
